__tests__/util/adminApi/auth.test.tsx

```diff
@@ -0,0 +1,183 @@
+import { StatusCodes } from "http-status-codes";
+import prisma from "../../../lib/services/db";
+import { authenticate } from "../../../util/adminApi/auth";
+import { decodeToken } from "../../../util/adminApi/tokenDecoding";
+
+jest.mock("../../../lib/services/db", () => ({
+  organisationAdminToken: {
+    findFirst: jest.fn(),
+  },
+  appAdminToken: {
+    findFirst: jest.fn(),
+  },
+}));
+
+jest.mock("../../../util/logger", () => ({
+  Logger: jest.fn().mockImplementation(() => ({
+    log: jest.fn(),
+    error: jest.fn(),
+  })),
+}));
+
+jest.mock("../../../util/adminApi/tokenDecoding", () => ({
+  decodeToken: jest.fn(),
+}));
+
+const appPrefix = "app";
+const orgPrefix = "org";
+const testToken = "abcdefgh1234567890";
+
+describe("Test auth for admin api tokens", () => {
+  it("authenticates app admin api token successfully", async () => {
+    // -- Arrange --
+    const req = { headers: { authorization: "Bearer app_abcdefgh1234567890" } };
+
+    (decodeToken as jest.Mock).mockReturnValue({
+      token: testToken,
+      type: appPrefix,
+    });
+    (prisma.appAdminToken.findFirst as jest.Mock).mockResolvedValue({
+      appId: 123,
+      token: testToken,
+      isDeleted: false,
+    });
+
+    // -- Act  --
+    const result = await authenticate(req as any, "app");
+
+    // -- Assert  --
+    expect(result).toEqual({
+      success: true,
+      authToken: testToken,
+      id: 123,
+      statusCode: StatusCodes.OK,
+    });
+  });
+
+  it("authenticates org admin api token successfully", async () => {
+    // -- Arrange --
+    const req = { headers: { authorization: "Bearer org_abcdefgh1234567890" } };
+
+    (decodeToken as jest.Mock).mockReturnValue({
+      token: testToken,
+      type: orgPrefix,
+    });
+    (prisma.organisationAdminToken.findFirst as jest.Mock).mockResolvedValue({
+      orgId: 123,
+      token: testToken,
+      isDeleted: false,
+    });
+
+    // -- Act  --
+    const result = await authenticate(req as any, "org");
+
+    // -- Assert  --
+    expect(result).toEqual({
+      success: true,
+      authToken: testToken,
+      id: 123,
+      statusCode: StatusCodes.OK,
+    });
+  });
+
+  it("authenticates org admin api token without Bearer prefix successfully", async () => {
```

Comment:
Is there a specific reason we are supporting with and without prefix?
If we support without the prefix, why should we support having the prefix?